airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/buffers/BufferDequeue.java

```diff
@@ -76,6 +78,12 @@ public MemoryAwareMessageBatch take(final StreamDescriptor streamDescriptor, fin
 
       queue.addMaxMemory(-bytesRead.get());
 
+      if (buffers.get(streamDescriptor).isEmpty()) {
+        final long remainingBlockSize = BLOCK_SIZE_BYTES - bytesRead.get() % BLOCK_SIZE_BYTES;
+        bytesRead.addAndGet(remainingBlockSize);
+        buffers.remove(streamDescriptor);
+      }
```

Comment:
I think this should more or less do it, but I have a couple nits.

Since we have the globalMemoryManager here and we are removing the buffer, it's probably better to directly "free" the slop memory here, instead of adding it to the memory for the batch to release later. This should make the logic and logs clearer and also speed up releasing of the memory.

We also probably want to add a comment at least.